contrib/hashicorp/vault/vault_test.go

```diff
@@ -0,0 +1,440 @@
+		t.Fatal(err)
+	}
+	defer mountKV(client, t)()
+
+	key := "/some/bad/key"
+	fullPath := "/v1" + key
+	secret, err := client.Logical().Read(key)
+	if err == nil {
+		t.Fatalf("Expected error when reading key from %s, but it returned: %#v", key, secret)
+	}
+	spans := mt.FinishedSpans()
+	assert.Len(spans, 2)
+	span := spans[1]
+
+	// Read key error
+	assert.Equal("vault", span.Tag(ext.ServiceName))
+	assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+	assert.Equal(http.MethodGet, span.Tag(ext.HTTPMethod))
+	assert.Equal(http.MethodGet+" "+fullPath, span.Tag(ext.ResourceName))
+	assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+	assert.Equal(404, span.Tag(ext.HTTPCode))
+	assert.Equal(true, span.Tag(ext.Error))
+	assert.NotZero(span.Tag(ext.ErrorMsg))
+	assert.Zero(span.Tag("vault.namespace"))
+}
+
+func TestNamespace(t *testing.T) {
+	assert := assert.New(t)
+
+	ts, cleanupTs := setupServer(t)
+	defer cleanupTs()
+	client, err := setupClient(ts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mountKV(client, t)()
+
+	namespace := "/some/namespace"
+	client.SetNamespace(namespace)
+	key := secretMountPath + "/testNamespace"
+	fullPath := "/v1" + key
+
+	t.Run("write", func(t *testing.T) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		// Write key with namespace
+		data := map[string]interface{}{"Key1": "Val1", "Key2": "Val2"}
+		_, err = client.Logical().Write(key, data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		spans := mt.FinishedSpans()
+		assert.Len(spans, 1)
+		span := spans[0]
+
+		assert.Equal("vault", span.Tag(ext.ServiceName))
+		assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+		assert.Equal(http.MethodPut, span.Tag(ext.HTTPMethod))
+		assert.Equal(http.MethodPut+" "+fullPath, span.Tag(ext.ResourceName))
+		assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+		assert.Equal(200, span.Tag(ext.HTTPCode))
+		assert.Zero(span.Tag(ext.Error))
+		assert.Zero(span.Tag(ext.ErrorMsg))
+		assert.Equal(namespace, span.Tag("vault.namespace"))
+	})
+
+	t.Run("read", func(t *testing.T) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		// Write key with namespace first
+		data := map[string]interface{}{"Key1": "Val1", "Key2": "Val2"}
+		_, err = client.Logical().Write(key, data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		// Read key with namespace
+		_, err = client.Logical().Read(key)
+		if err != nil {
+			t.Fatal(err)
+		}
+		spans := mt.FinishedSpans()
+		assert.Len(spans, 2)
+		span := spans[1]
+
+		assert.Equal("vault", span.Tag(ext.ServiceName))
+		assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+		assert.Equal(http.MethodGet, span.Tag(ext.HTTPMethod))
+		assert.Equal(http.MethodGet+" "+fullPath, span.Tag(ext.ResourceName))
+		assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+		assert.Equal(200, span.Tag(ext.HTTPCode))
+		assert.Zero(span.Tag(ext.Error))
+		assert.Zero(span.Tag(ext.ErrorMsg))
+		assert.Equal(namespace, span.Tag("vault.namespace"))
+	})
+}
+
+func getWriteSpan(t *testing.T, conf *api.Config) mocktracer.Span {
```

Comment:
Please [don't use `get*` in Go](https://golang.org/doc/effective_go.html#Getters).

Reply:
I read that section as not to use `get` in the name of a getter, rather than not to use `get` in the beginning of any function, but this is now gone at any rate.

Reply:
I see. I frankly don't know how to define a getter. I just have a reflex to make this comment whenever I see `get*` and avoid using it overall. I'm sure there are exceptions when it's ok to use `get`.